src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -256,18 +323,29 @@ public void AssignLayers()
                     layerWidth = 0;
 
                     if (n.LeftEdges.Count == 0)
+                    {
                         isFinalLayer = true;
+                    }
                 }
 
                 AddToLayer(n, currentLayer);
                 processed++;
 
                 if (n.Width > layerWidth)
+                {
                     layerWidth = n.Width;
+                }
             }
 
-            // Put the input nodes on the leftmost layer
-            AddToLayer(Nodes.Where(x => x.LeftEdges.Count == 0).ToList(), Layers.Count);
+            // Put all input nodes and isolated nodes on the leftmost layer
+            AddToLayer(Nodes.Where(x =>
+                (x.LeftEdges.Count == 0 && !x.HasUnconnectedInPort) ||
```

Comment:
Does `LeftEdges` signify the connectors going into a node as input? If so, I'm unable to see how a node can have no `LeftEdges` and yet have a connected input port?